Alignment/HIPAlignmentAlgorithm/plugins/HIPTwoBodyDecayAnalyzer.cc

```diff
@@ -0,0 +1,385 @@
+  }
+  shortBranches.clear();
+  for (unsigned int el=0; el<intBranches.size(); el++){
+    if (intBranches.at(el).second!=0) delete intBranches.at(el).second;
+    intBranches.at(el).second=0;
+  }
+  intBranches.clear();
+  for (unsigned int el=0; el<floatBranches.size(); el++){
+    if (floatBranches.at(el).second!=0) delete floatBranches.at(el).second;
+    floatBranches.at(el).second=0;
+  }
+  floatBranches.clear();
+}
+void HIPTwoBodyDecayAnalyzer::initializeBranches(){
+  for (unsigned short el=0; el<shortBranches.size(); el++){
+    if (shortBranches.at(el).second!=0) *(shortBranches.at(el).second)=0;
+  }
+  for (unsigned int el=0; el<intBranches.size(); el++){
+    if (intBranches.at(el).second!=0) *(intBranches.at(el).second)=0;
+  }
+  for (unsigned int el=0; el<floatBranches.size(); el++){
+    if (floatBranches.at(el).second!=0) *(floatBranches.at(el).second)=0;
+  }
+}
+
+void HIPTwoBodyDecayAnalyzer::bookAllBranches(){
+  const int nTrackTypes = 4;
+  TString strTrackTypes[nTrackTypes]={
+    "alcareco",
+    "refit1",
+    "refit2",
+    "final"
+  };
+  for (unsigned int it=0; it<4; it++){
+    TString strTrackType = strTrackTypes[it];
+    bookBranch(strTrackType + "_present", "s");
+    bookBranch(strTrackType + "_ZVtxFitOk", "s");
+    bookBranch(strTrackType + "_ZMass", "f"); bookBranch(strTrackType + "_ZPt", "f"); bookBranch(strTrackType + "_ZPz", "f"); bookBranch(strTrackType + "_ZPhi", "f");
+    bookBranch(strTrackType + "_ZVertex_x", "f"); bookBranch(strTrackType + "_ZVertex_y", "f"); bookBranch(strTrackType + "_ZVertex_z", "f"); bookBranch(strTrackType + "_ZVertex_NormChi2", "f");
+    bookBranch(strTrackType + "_MuPlusVertex_x", "f"); bookBranch(strTrackType + "_MuPlusVertex_y", "f"); bookBranch(strTrackType + "_MuPlusVertex_z", "f");
+    bookBranch(strTrackType + "_MuMinusPt_AfterZVtxFit", "f"); bookBranch(strTrackType + "_MuMinusPz_AfterZVtxFit", "f"); bookBranch(strTrackType + "_MuMinusPhi_AfterZVtxFit", "f");
+    bookBranch(strTrackType + "_MuPlusPt_AfterZVtxFit", "f"); bookBranch(strTrackType + "_MuPlusPz_AfterZVtxFit", "f"); bookBranch(strTrackType + "_MuPlusPhi_AfterZVtxFit", "f");
+    bookBranch(strTrackType + "_ZMass_AfterZVtxFit", "f"); bookBranch(strTrackType + "_ZPt_AfterZVtxFit", "f"); bookBranch(strTrackType + "_ZPz_AfterZVtxFit", "f"); bookBranch(strTrackType + "_ZPhi_AfterZVtxFit", "f");
+    bookBranch(strTrackType + "_MuMinusPt", "f"); bookBranch(strTrackType + "_MuMinusPz", "f"); bookBranch(strTrackType + "_MuMinusPhi", "f");
+    bookBranch(strTrackType + "_MuMinusVertex_x", "f"); bookBranch(strTrackType + "_MuMinusVertex_y", "f"); bookBranch(strTrackType + "_MuMinusVertex_z", "f");
+    bookBranch(strTrackType + "_MuPlusPt", "f"); bookBranch(strTrackType + "_MuPlusPz", "f"); bookBranch(strTrackType + "_MuPlusPhi", "f");
+  }
+  actuateBranches();
+}
+bool HIPTwoBodyDecayAnalyzer::bookBranch(TString bname, TString btype){
+  if (btype=="f") floatBranches.emplace_back(bname, new Float_t);
```

Comment:
I could do this, but again per my comment above, I would prefer to do this in a separate PR if possible.